### PR TITLE
eval_downstream support for go_emotions dataset

### PR DESCRIPTION
**Pull request by bfineran**
**test_plan:**
tested against sample model

### CODE REVIEW

**Review by KSGulin — APPROVED** [review]
LGTM

File: src/deepsparse/transformers/eval_downstream.py [hunk 1]
```diff
@@ -365,6 +367,48 @@ def conll2003_eval(args):
...
+        model_path=args.onnx_filepath,
+        engine_type=args.engine,
+        num_cores=args.num_cores,
+        sequence_length=args.max_sequence_length,
+        top_k=num_labels,
+    )
+    print(f"Engine info: {text_classify.engine}")
+
+    go_emotions_metrics = PrecisionRecallF1(id_to_label=text_classify.config.id2label)
+    ordered_labels = [text_classify.config.id2label[idx] for idx in range(num_labels)]
```
Comment (rahul-tuli) [inline src/deepsparse/transformers/eval_downstream.py:387]: I get an key error on this line with the following command:

```bash
deepsparse.transformers.eval_downstream \
"zoo:nlp/sentiment_analysis/bert-base/pytorch/huggingface/sst2/12layer_pruned80_quant-none-vnni" \
--dataset go_emotions
```

Traceback:
```bash
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 612, in <module>
    main()
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 608, in main
    _main(args)
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 593, in _main
    metrics = SUPPORTED_DATASETS[dataset](args)
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 387, in go_emotions_eval
    ordered_labels = [text_classify.config.id2label[idx] for idx in range(num_labels)]
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 387, in <listcomp>
    ordered_labels = [text_classify.config.id2label[idx] for idx in range(num_labels)]
KeyError: 2

Process finished with exit code 1
```

this is tested on current branch with the following transformers:
```bash
Name: transformers
Version: 4.23.1
Summary: State-of-the-art Machine Learning for JAX, PyTorch and TensorFlow
Home-page: https://github.com/huggingface/transformers
Author: The Hugging Face team (past and future) with the help of all our contributors (https://github.com/huggingface/transformers/graphs/contributors)
Author-email: transformers@huggingface.co
License: Apache
Location: /home/XXXXX/virtual_environments/deepsparse3.8/lib/python3.8/site-packages
Requires: filelock, huggingface-hub, numpy, packaging, pyyaml, regex, requests, tokenizers, tqdm
Required-by: farm-haystack, sentence-transformers
```

Could we re-verify before landing?

**Review by rahul-tuli — CHANGES_REQUESTED** [review]
Code looks great, called out a potential bug, looks good to land from my side once bug is resolved/verified

File: src/deepsparse/transformers/eval_downstream.py (hunk 1 above)
Comment (bfineran) [reply src/deepsparse/transformers/eval_downstream.py:387]: The model you are using is trained for sst2, not go emotions so is failing due to a label mismatch

**Comment by bfineran** [comment]
Responded. Need to investigate test failures

**Review by rahul-tuli — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]

**Pull request merged**